src/Vivait/LicensingClientBundle/Controller/TokenController.php

```diff
@@ -25,9 +25,9 @@ public function tokenAction(Request $request)
 
         try {
             $tokenData = $licensingApi->getToken(
-                $request->query->get('client_id', null),
-                $request->query->get('client_secret', null),
-                $request->query->get('grant_type', 'client_credentials')
+                $request->request->get('client_id', $request->query->get('client_id', null)),
```

Comment:
You can do this directly on the request and it will [try the GET, then the POST etc.](http://api.symfony.com/2.7/Symfony/Component/HttpFoundation/Request.html#method_get).